# Project - EDA with Pandas Using the Boston Housing Data

## Introduction

In this section you've learned a lot about importing, cleaning up, analyzing (using descriptive statistics) and visualizing data. In this a more free form project you'll get a chance to practice all of these skills with the Boston Housing data set, which contains housing values in suburbs of Boston. The Boston Housing Data is commonly used by aspiring data scientists.

In [1]:
import pandas as pd
import numpy as np

## Objectives

You will be able to:

* Load csv files using Pandas
* Find variables with high correlation
* Create box plots

## Goals

Use your data munging and visualization skills to conduct an exploratory analysis of the dataset below. At a minimum, this should include:

* Loading the data (which is stored in the file `train.csv`)

In [2]:
df = pd.read_csv('train.csv')

* Use built-in python functions to explore measures of centrality and dispersion for at least 3 variables

In [3]:
df.info

<bound method DataFrame.info of       ID      crim    zn  indus  chas    nox     rm    age     dis  rad  tax  \
0      1   0.00632  18.0   2.31     0  0.538  6.575   65.2  4.0900    1  296   
1      2   0.02731   0.0   7.07     0  0.469  6.421   78.9  4.9671    2  242   
2      4   0.03237   0.0   2.18     0  0.458  6.998   45.8  6.0622    3  222   
3      5   0.06905   0.0   2.18     0  0.458  7.147   54.2  6.0622    3  222   
4      7   0.08829  12.5   7.87     0  0.524  6.012   66.6  5.5605    5  311   
5     11   0.22489  12.5   7.87     0  0.524  6.377   94.3  6.3467    5  311   
6     12   0.11747  12.5   7.87     0  0.524  6.009   82.9  6.2267    5  311   
7     13   0.09378  12.5   7.87     0  0.524  5.889   39.0  5.4509    5  311   
8     14   0.62976   0.0   8.14     0  0.538  5.949   61.8  4.7075    4  307   
9     15   0.63796   0.0   8.14     0  0.538  6.096   84.5  4.4619    4  307   
10    16   0.62739   0.0   8.14     0  0.538  5.834   56.5  4.4986    4  307   
11    17

* Create *meaningful* subsets of the data using selection operations using `.loc`, `.iloc` or related operations. Explain why you used the chosen subsets and do this for 3 possible 2-way splits. State how you think the 2 measures of centrality and/or dispersion might be different for each subset of the data. Examples of potential splits:

    - Create a 2 new dataframes based on your existing data, where one contains all the properties next to the Charles river, and the other one contains properties that aren't.
    - Create 2 new dataframes based on a certain split for crime rate.

* Next, use histograms and scatterplots to see whether you observe differences for the subsets of the data. Make sure to use subplots so it is easy to compare the relationships.

## Variable Descriptions

This data frame contains the following columns:

#### crim  
per capita crime rate by town.

#### zn  
proportion of residential land zoned for lots over 25,000 sq.ft.

#### indus  
proportion of non-retail business acres per town.

#### chas  
Charles River dummy variable (= 1 if tract bounds river; 0 otherwise).

#### nox  
nitrogen oxides concentration (parts per 10 million).

#### rm  
average number of rooms per dwelling.

#### age  
proportion of owner-occupied units built prior to 1940.

#### dis  
weighted mean of distances to five Boston employment centers.

#### rad  
index of accessibility to radial highways.

#### tax  
full-value property-tax rate per $10,000.

#### ptratio  
pupil-teacher ratio by town.

#### black  
1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town.

#### lstat  
lower status of the population (percent).

#### medv  
median value of owner-occupied homes in $10000s.
  
  
  
Source
Harrison, D. and Rubinfeld, D.L. (1978) Hedonic prices and the demand for clean air. J. Environ. Economics and Management 5, 81–102.

Belsley D.A., Kuh, E. and Welsch, R.E. (1980) Regression Diagnostics. Identifying Influential Data and Sources of Collinearity. New York: Wiley.

## Summary

Congratulations, you've completed your first "freeform" exploratory data analysis of a popular data set!